src/libraries/System.ComponentModel.Annotations/src/System/ComponentModel/DataAnnotations/EnumDataTypeAttribute.cs

```diff
@@ -35,7 +35,7 @@ public override bool IsValid(object? value)
                 return true;
             }
             var stringValue = value as string;
-            if (stringValue?.Length == 0)
+            if (stringValue != null && stringValue.Length == 0)
```

Comment:
`?.Length == 0` strictly equals to `!= null` and `Length == 0`. This doesn't change anything.
This case is intentional to *not* handle null. It can also be simple written as `stringValue == ""`.